test/known_problems/should_fail/return_fun_should_fail.erl

```diff
@@ -0,0 +1,8 @@
+-module(return_fun_should_fail).
+
+-export([return_fun_no_spec/0]).
+
+-spec return_fun_no_spec() -> integer().
+return_fun_no_spec() -> fun no_spec/0.
```

Comment:
This can be solved by this commit https://github.com/xxdavid/Gradualizer/commit/13d689363e96f0c4f74ebc70b292ac43657a3540
You can cherry pick it if you want.